hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/DatanodeChunkGenerator.java

```diff
@@ -95,45 +95,47 @@ public Void call() throws Exception {
           "Datanode chunk generator is not supported in secure environment");
     }
 
-    StorageContainerLocationProtocol scmLocationClient =
-        createStorageContainerLocationClient(ozoneConf);
-    List<Pipeline> pipelines = scmLocationClient.listPipelines();
-
-    Pipeline pipeline;
-    if (pipelineId != null && pipelineId.length() > 0) {
-      pipeline = pipelines.stream()
-          .filter(p -> p.getId().toString().equals(pipelineId))
-          .findFirst()
-          .orElseThrow(() -> new IllegalArgumentException(
-              "Pipeline ID is defined, but there is no such pipeline: "
-                  + pipelineId));
-
-    } else {
-      pipeline = pipelines.stream()
-          .filter(p -> p.getFactor() == ReplicationFactor.THREE)
-          .findFirst()
-          .orElseThrow(() -> new IllegalArgumentException(
-              "Pipeline ID is NOT defined, and no pipeline has been found with "
-                  + "factor=THREE"));
-      LOG.info("Using pipeline {}", pipeline.getId().toString());
+    try (StorageContainerLocationProtocol scmLocationClient =
+        createStorageContainerLocationClient(ozoneConf)) {
+      List<Pipeline> pipelines = scmLocationClient.listPipelines();
+      Pipeline pipeline;
+      if (pipelineId != null && pipelineId.length() > 0) {
+        pipeline = pipelines.stream()
+            .filter(p -> p.getId().toString().equals(pipelineId))
+            .findFirst()
+            .orElseThrow(() -> new IllegalArgumentException(
+                "Pipeline ID is defined, but there is no such pipeline: "
+                    + pipelineId));
 
-    }
-
-    xceiverClientSpi =
-        new XceiverClientManager(ozoneConf).acquireClient(pipeline);
+      } else {
+        pipeline = pipelines.stream()
+            .filter(p -> p.getFactor() == ReplicationFactor.THREE)
+            .findFirst()
+            .orElseThrow(() -> new IllegalArgumentException(
+                "Pipeline ID is NOT defined, and no pipeline " +
+                    "has been found with factor=THREE"));
+        LOG.info("Using pipeline {}", pipeline.getId());
+      }
 
-    timer = getMetrics().timer("chunk-write");
+      xceiverClientSpi =
+          new XceiverClientManager(ozoneConf).acquireClient(pipeline);
```

Comment:
Turns out that in addition to the client, the `XceiverClientManager` should be closed, too.

Reply:
Good catch! Will fix it in the next patch!